sys/mips/malta/malta_machdep.c

```diff
@@ -369,19 +438,34 @@ platform_start(__register_t a0, __register_t a1,  __register_t a2,
 	cninit();
 	printf("entry: platform_start()\n");
 
-	bootverbose = 1;
-
-	/* 
-	 * YAMON uses 32bit pointers to strings so
-	 * sign-extend them to the correct type manually.
+	/*
+	 * Parse kernel cmdline.
 	 */
-
-	if (bootverbose) {
+	if (bootverbose)
 		printf("cmd line: ");
-		for (i = 0; i < argc; i++)
-			printf("'%s' ", (char*)(intptr_t)argv[i]);
-		printf("\n");
+	for (i = 0; i < argc; i++) {
+		char *arg = platform_argv_ptr(argv[i]);
+		if (bootverbose)
+			printf("%s ", arg);
+		while (*arg != '\0') {
+			if (*arg++ != '-')
+				continue;
+			switch (*arg) {
+			case 'a':
+				boothowto |= RB_ASKNAME;
+				break;
+			case 'v':
+				bootverbose = 1;
+				boothowto |= RB_VERBOSE;
+				break;
+			case 'C':
+				boothowto |= RB_CDROM;
+				break;
+			}
+		}
 	}
+	if (bootverbose)
+		printf("\n");
```

Comment:
None of this should be changing other than the pointer passed to printf

Reply:
IIRC I changed that to be able to boot MIPS from an iso9660 filesystem cdrom device when UFS was not working yet. I can yank it out.

Reply:
I think that would be best